File: dtc-flink-zabbix/src/main/scala/com/dtc/scala/analytic/common/Count.scala
package com.dtc.scala.analytic.common

import java.util.concurrent.atomic.AtomicLong

import scala.collection.mutable

/**
  * Created on 2019-06-12
  *
  * @author :hao.li
  */
class Count {
  var counterMap= new mutable.HashMap[String,AtomicLong]
  var atomicLong = new AtomicLong(0)
  val COUNTER_EVENTS_RECEIVED: String = "counter.events.received"
  val COUNTER_EVENTS_ERROR: String = "counter.events.error"
  val COUNTER_EVENTS_RIGHT: String = "counter.events.right"

  def increment(counter:String): Long ={
    if(counterMap.get(counter).isEmpty){
      counterMap.put(counter,atomicLong)
      return 0L
    }else{
      return counterMap(counter).incrementAndGet()
    }

  }
  def getCount(counter:String):Long={
    return counterMap(counter).toString.toLong
  }

}

File: dtc-flink-zabbix/src/main/scala/com/dtc/java/analytic/V1/common/utils/Count.java
package com.dtc.java.analytic.V1.common.utils;

import org.apache.directory.api.util.Strings;

import java.util.HashMap;
import java.util.Map;
import java.util.concurrent.atomic.AtomicLong;

/**
 * Created on 2019-08-21
 * statistic the data of recive and error data
 *
 * @author :hao.li
 */
public class Count {
    Map<String, AtomicLong> counterMap = new HashMap<>();
    AtomicLong atomicLong = new AtomicLong(0);

    public Long increment(String counts) {
        AtomicLong atomicLong = counterMap.get(counts);
        if (atomicLong == null) {
            counterMap.put(counts, this.atomicLong);
            return 0L;
        } else {
            return counterMap.get(counts).incrementAndGet();
        }
    }

    public Long getCount(String counts) {
        return counterMap.get(counts).get();
    }
}
